File: src/main/java/com/sjs/lootbotga/game/cards/PirateCardBuilder.java
package com.sjs.lootbotga.game.cards;

public class PirateCardBuilder {
    private FleetType fleetType;
    private PirateValue pirateValue;

    public PirateCardBuilder fleet(FleetType fleetType) {
        this.fleetType = fleetType;
        return this;
    }

    public PirateCardBuilder value(PirateValue pirateValue) {
        this.pirateValue = pirateValue;
        return this;
    }

    public final Card build() {
        return new Card(CardType.PIRATE, fleetType, pirateValue);
    }
}

File: src/main/kotlin/com/sjs/lootbotga/game/cards/PirateCardBuilder.kt
package com.sjs.lootbotga.game.cards

data class PirateCardBuilder(private val fleetType: FleetType, private val pirateValue: PirateValue) : CardBuilder {
    override fun build(): Card {
        return Card(CardType.PIRATE, fleetType, pirateValue)
    }
}